### PR TITLE
feat(onboarding): add datasource sysdig_secure_trusted_azure_app

### PR DESCRIPTION
**Pull request by cgeers**

### CODE REVIEW

File: sysdig/internal/client/v2/onboarding.go [hunk 1]
```diff
@@ -8,15 +8,17 @@ import (
...
 	onboardingAgentlessScanningAssetsPath = "%s/api/secure/onboarding/v2/agentlessScanningAssets"
 )
 
 type OnboardingSecureInterface interface {
 	Base
 	GetTrustedCloudIdentitySecure(ctx context.Context, provider string) (string, error)
+	GetTrustedAzureAppSecure(ctx context.Context, app string) (map[string]string, error)
 	GetTenantExternalIDSecure(ctx context.Context) (string, error)
-	GetAgentlessScanningAssetsSecure(ctx context.Context) (map[string]interface{}, error)
+	GetAgentlessScanningAssetsSecure(ctx context.Context) (map[string]any, error)
```
Comment (ravinadhruve10) [inline sysdig/internal/client/v2/onboarding.go:21]: any reason for changing this return type?

File: website/docs/d/secure_trusted_azure_app.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+## Attributes Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+* `tenant_id` - The application's associated tenant identifer
+
+* `application_id` - The application's identifier
+
+* `service_principal_id` - The application's assocaited service principal identifier.
```
Comment (ravinadhruve10) [inline website/docs/d/secure_trusted_azure_app.md:36]: nit: associated

File: sysdig/data_source_sysdig_secure_onboarding.go [hunk 3]
```diff
@@ -94,6 +96,62 @@ func dataSourceSysdigSecureTrustedCloudIdentityRead(ctx context.Context, d *sche
...
+		return diag.FromErr(err)
+	}
+
+	app := d.Get("name").(string)
+	registration, err := client.GetTrustedAzureAppSecure(ctx, app)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	d.SetId(app)
+	for k, v := range registration {
```
Comment (ravinadhruve10) [inline sysdig/data_source_sysdig_secure_onboarding.go:144]: This depends on the approach chosen for config in the backend PR. Rest lgtm!

File: sysdig/internal/client/v2/onboarding.go (hunk 1 above)
Comment (cgeers) [reply sysdig/internal/client/v2/onboarding.go:21]: it's not really a change, it's the preferred nomenclature

File: sysdig/data_source_sysdig_secure_onboarding.go (hunk 3 above)
Comment (cgeers) [reply sysdig/data_source_sysdig_secure_onboarding.go:144]: I'm not sure I understand this comment.

File: website/docs/d/secure_trusted_azure_app.md (hunk 2 above)
Comment (cgeers) [reply website/docs/d/secure_trusted_azure_app.md:36]: fixed.

File: sysdig/data_source_sysdig_secure_onboarding.go (hunk 3 above)
Comment (ravinadhruve10) [reply sysdig/data_source_sysdig_secure_onboarding.go:144]: Had commented this 2 days ago, while we were discussing the approach of how we want to store the env config and pass them as response to the api. Left this comment if based on the approach decided, this reading of api response needs to change. It doesn't need to, so this comment can be ignored/resolved.

**Review by ravinadhruve10 — APPROVED** [review]

**Pull request merged**